clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunction.cpp

```diff
@@ -187,16 +261,13 @@ void createCoercedStore(mlir::Value Src, mlir::Value Dst, bool DstIsVolatile,
     auto addr = bld.create<CastOp>(Dst.getLoc(), ptrTy, CastKind::bitcast, Dst);
     bld.create<StoreOp>(Dst.getLoc(), Src, addr);
   } else {
-    cir_cconv_unreachable("NYI");
+    auto tmp = createTmpAlloca(CGF, Src.getLoc(), SrcTy);
+    CGF.getRewriter().create<StoreOp>(Src.getLoc(), Src, tmp);
+    auto len = createUInt64(CGF, Src.getLoc(), DstSize.getFixedValue());
+    createMemCpy(CGF, tmp, Dst, len);
```

Comment:
here are the changes for the title - we emit a temp alloca (basically that's why we need the changes in the `CIRBaseBuilder` - we don't have any hint here where to insert it. Thus, we insert it as LLVM does - in the function start